src/pages/chainlink-nodes/configuring-nodes.mdx

```diff
@@ -0,0 +1,127 @@
+
+Before you begin, update your Chainlink Node to v1.13.0 and ensure that it operates as expected. This is the last version that supports environment variable configuration, and the best version to use for TOML migration before you update the node to future versions.
+
+### Export your current config
+
+You can generate the `config.toml` file using the `chainlink config dump` command. This guide shows you how to do this for a Chainlink node running in a Docker container:
+
+1. Open an interactive shell inside the Docker container that runs your node:
+
+   ```shell
+   docker exec -it chainlink bash
+   ```
+
+1. Log in to the Chainlink CLI with an account that has admin access:
+
+   ```shell
+   chainlink admin login
+   ```
+
+1. Export the current config to a `config.toml` file:
+
+   ```shell
+   chainlink config dump > config.toml
+   ```
+
+1. Log out of the Chainlink CLI and close the shell on the Docker container.
+
+   ```shell
+   chainlink admin logout && exit
+   ```
+
+1. Make a new directory on the host system to store your config files. Then, `cd` to that directory. This directory will be mounted to Docker when you run the node using the TOML config.
+
+   ```shell
+   mkdir ~/.chainlink-toml && cd ~/.chainlink-toml
+   ```
+
+1. Write the config file from the container to your host:
+
+   ```shell
+   docker exec -it chainlink cat /home/chainlink/config.toml > ./config.toml
+   ```
+
+1. Create a `secrets.toml` file with the minimum required secrets, which are `[Database]` and `[Password]`. If you are working on a test node, you can use `?sslmode=disable` in the database `URL`:
+
+   ```shell
+   echo "[Database]
+   URL = 'postgresql://user:pass@localhost:5432/dbname'
+
+   [Password]
+   Keystore = 'keystore_pass'" > ./secrets.toml
+   ```
+
+1. Edit the `secrets.toml` file to include the secrets you need for your specific Chainlink node. See the [Node Secrets](/chainlink-nodes/v1/secrets-config) page for a full list of available config options.
+
+### Validate the configuration
+
+After you have the `config.toml` file and the `secrets.toml` files on the host system, you can validate these files using a temporary Docker container.
+
+1. Validate the configuration:
+
+   ```shell
+   docker run --name chainlink-config-validator -v ~/.chainlink-toml:/chainlink -it --rm smartcontract/chainlink:1.13.0 -config /chainlink/config.toml -secrets /chainlink/secrets.toml config validate
+   ```
+
+   You will likely see some invalid config errors. For example:
+
+   ```shell
+   Invalid configuration: invalid configuration: EVM.3.Nodes: missing: must have at least one node
+   ```
+
+1. Edit the `config.toml` file and manually correct any errors. See the [Node Config](/chainlink-nodes/v1/node-config) and [Node Secrets](/chainlink-nodes/v1/secrets-config) pages to learn which config settings are valid. For the error in this example, an EVM chain was configured with no nodes. Removing this from the config made the config valid:
+
```

Comment:
The examples at the top of the config and secrets docs are meant to be minimalish. There are only five total required fields (for evm anyways), although `WSURL` is also conditionally required since with one node it can't be send-only.
https://github.com/smartcontractkit/chainlink/blob/develop/docs/CONFIG.md#example
https://github.com/smartcontractkit/chainlink/blob/develop/docs/SECRETS.md#example

Reply:
> is there are a way to generate a TOML dump will all the defaults omitted, so that you can see which fields have been explicitly configured

What do you mean? This is how the literal `dump` sub-command works, and the logs on startup distinguish input from effective (w/ defaults).

Reply:
The [Running a Chainlink Node](https://documentation-git-fork-dwightjl-main-chainlinklabs.vercel.app/chainlink-nodes/v1/running-a-chainlink-node#run-chainlink-node) guide has a minimal config to help users get started. Does that address your request here?

Reply:
>What do you mean? This is how the literal dump sub-command works, and the logs on startup distinguish input from effective (w/ defaults).

in this doc, the user is instructed the run a `chainlink-validator` docker image. it prints out the entire config, including defaults, which is overwhelming

Reply:
> The [Running a Chainlink Node](https://documentation-git-fork-dwightjl-main-chainlinklabs.vercel.app/chainlink-nodes/v1/running-a-chainlink-node#run-chainlink-node) guide has a minimal config to help users get started. Does that address your request here?

yes, i missed the link in my first reading

Reply:
> in this doc, the user is instructed the run a chainlink-validator docker image. it prints out the entire config, including defaults, which is overwhelming

It also prints out the input separate from the full effective, just like when starting the node:
https://github.com/smartcontractkit/chainlink/blob/489280eb6530092afe8711ca059d201d7091e97e/core/services/chainlink/config_general.go#L238-L241

Reply:
PR approved from krehermann. Resolving this thread.